saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx

```diff
@@ -0,0 +1,513 @@
+import { UpdateShippingZone } from "../../types/UpdateShippingZone";
+import { shippingZonesListUrl, shippingZoneUrl } from "../../urls";
+import ShippingZoneOperations from "./ShippingZoneOperations";
+import {
+  shippingZoneAssignCountryPath,
+  shippingZoneAssignCountryUrl,
+  shippingZoneDeletePath,
+  shippingZoneDeleteUrl,
+  shippingZonePriceRateCreatePath,
+  shippingZonePriceRateCreateUrl,
+  shippingZoneRateDeletePath,
+  shippingZoneRateDeleteUrl,
+  shippingZoneRatePath,
+  shippingZoneRateUrl,
+  shippingZoneUnassignCountryPath,
+  shippingZoneUnassignCountryUrl,
+  shippingZoneWeightRateCreatePath,
+  shippingZoneWeightRateCreateUrl
+} from "./urls";
+
+export interface ShippingZoneDetailsProps {
+  id: string;
+}
+
+const ShippingZoneDetails: React.StatelessComponent<
+  ShippingZoneDetailsProps
+> = ({ id }) => (
+  <Navigator>
+    {navigate => (
+      <Messages>
+        {pushMessage => (
+          <Shop>
+            {shop => {
+              const closeModal = () => navigate(shippingZoneUrl(id));
+
+              const onShippingRateCreate = (data: CreateShippingRate) => {
+                if (data.shippingPriceCreate.errors.length === 0) {
+                  pushMessage({
+                    text: i18n.t("Successfully created rate", {
+                      context: "shipping method"
+                    })
+                  });
+                  closeModal();
+                }
+              };
+
+              const onShippingRateUpdate = (data: UpdateShippingRate) => {
+                if (data.shippingPriceUpdate.errors.length === 0) {
+                  pushMessage({
+                    text: i18n.t("Successfully updated rate", {
+                      context: "shipping method"
+                    })
+                  });
+                  closeModal();
+                }
+              };
+
+              const onShippingRateDelete = (data: DeleteShippingRate) => {
+                if (data.shippingPriceDelete.errors.length === 0) {
+                  pushMessage({
+                    text: i18n.t("Successfully deleted rate", {
+                      context: "shipping method"
+                    })
+                  });
+                  closeModal();
+                }
+              };
+
+              const onShippingZoneDelete = (data: DeleteShippingZone) => {
+                if (data.shippingZoneDelete.errors.length === 0) {
+                  pushMessage({
+                    text: i18n.t("Successfully deleted shipping zone")
+                  });
+                  navigate(shippingZonesListUrl);
+                }
+              };
+
+              const onShippingZoneUpdate = (data: UpdateShippingZone) => {
+                if (data.shippingZoneUpdate.errors.length === 0) {
+                  pushMessage({
+                    text: i18n.t("Successfully updated shipping zone")
+                  });
+                  closeModal();
+                }
+              };
```

Comment:
DRY

Reply:
TBH I don't see any clean way to simplify this.

Reply:
Simple function which would accept key or `data.shippingZoneUpdate` etc